move/axelar_gateway/sources/types/weighted_signers.move

```diff
@@ -55,21 +72,21 @@ public(package) fun peel(bcs: &mut BCS): WeightedSigners {
 /// 2. The threshold is greater than zero.
 /// 3. The threshold is less than or equal to the total weight of the signers.
 public(package) fun validate(self: &WeightedSigners) {
-    assert!(!self.signers.is_empty(), EInvalidSigners);
-
-    validate_signers_order(self.signers);
-    let total_weight = calculate_total_weight(self.signers);
-    validate_threshold(self.threshold(), total_weight);
+    self.validate_signers();
+    self.validate_threshold();
 }
 
 /// Finds the weight of a signer in the weighted signers by its public key.
 public(package) fun find_signer_weight(
     self: &WeightedSigners,
+    signer_index: u64,
     pub_key: &vector<u8>,
-): u128 {
-    let mut signer = self.find!(|signer| signer.pub_key() == pub_key);
+): (u128, u64) {
+    // Find the signer by its public key
+    // The index of the signer is returned as well for reuse in the macro call
+    let (signer, index) = self.find!(signer_index, |signer| signer.pub_key() == pub_key);
```

Comment:
Don't think we need a macro here. It's not being reused anywhere else so it's just adding more complexity